integrations/datadog/queries.yaml

```diff
@@ -0,0 +1,184 @@
+cluster_replica_usage:
+  query: |
+          SELECT
+            U.replica_id,
+            R.cluster_id,
+            C.name as cluster_name,
+            R.name as replica_name,
+            U.cpu_percent,
+            U.memory_percent,
+            RS.credits_per_hour,
+            CASE
+              WHEN S.cluster_id IS NOT NULL THEN 'source'
+              WHEN SK.cluster_id IS NOT NULL THEN 'sink'
+              ELSE 'compute'
```

Comment:
Oh. Is there an `ingest clusters` type? Do we have a way to detect it? I missed it completely.

Reply:
There isn't, we're just encouraging users to create dedicated I/O clusters vs. using the option that creates linked clusters for each source or sink! Because you can't mix those objects of different types in the same cluster (sources/sinks, materialized views/indexes), you might have a cluster that is classified as `compute` but has multiple sources, sinks or both on it.

Reply:
I've tried the following and the source ended up with the correct label:

```sql
materialize=> CREATE CLUSTER safe_to_delete REPLICAS (
    r1 (SIZE = 'xsmall', INTROSPECTION INTERVAL = 0)
);
CREATE CLUSTER
materialize=> CREATE SOURCE counterd IN CLUSTER safe_to_delete 
  FROM LOAD GENERATOR COUNTER
  (TICK INTERVAL '500ms')
  ;
CREATE SOURCE
materialize=>             SELECT
              U.replica_id,
              R.cluster_id,
              C.name as cluster_name,
              R.name as replica_name,
              RST.status as replica_status_label,
              U.cpu_percent::float,
              U.memory_percent::float,
              RS.credits_per_hour::float,
              CASE RST.status
                WHEN 'ready' THEN 1
                ELSE 0
              END AS replica_status,
              CASE
                WHEN S.cluster_id IS NOT NULL THEN 'source'
                WHEN SK.cluster_id IS NOT NULL THEN 'sink'
                ELSE 'compute'
              END AS cluster_type
            FROM mz_internal.mz_cluster_replica_utilization U
            JOIN mz_catalog.mz_cluster_replicas R ON (U.replica_id = R.id)
            JOIN mz_catalog.mz_clusters C ON (R.cluster_id = C.id)
            JOIN mz_internal.mz_cluster_replica_sizes RS ON (R.size = RS.size)
            JOIN mz_internal.mz_cluster_replica_statuses RST ON (RST.replica_id = R.id)
            LEFT JOIN mz_catalog.mz_sources S ON (C.id = S.cluster_id)
            LEFT JOIN mz_catalog.mz_sinks SK ON (C.id = SK.cluster_id);
 replica_id | cluster_id |          cluster_name           | replica_name | replica_status_label |    cpu_percent     |   memory_percent    | credits_per_hour | replica_status | cluster_type 
------------+------------+---------------------------------+--------------+----------------------+--------------------+---------------------+------------------+----------------+--------------
 u96        | u86        | safe_to_delete                  | r1           | ready                |                    |                     |                1 |              1 | source
 u2         | s1         | mz_system                       | r1           | ready                |          1.4678427 |  0.2971649169921875 |              0.5 |              1 | compute
 u65        | u1         | default                         | xxsmall      | ready                | 3.0888546999999997 | 0.38352012634277344 |              0.5 |              1 | compute
 u89        | u79        | dogfood_ingest                  | r1           | ready                | 2.0672439000000002 |   0.383758544921875 |              0.5 |              1 | source
 u3         | s2         | mz_introspection                | r1           | ready                | 15.797568100000001 |   3.101825714111328 |              0.5 |              1 | compute
 u91        | u81        | ingest_instruqt                 | r1           | ready                | 1.9438331999999998 |  0.5001068115234375 |              0.5 |              1 | source
 u90        | u80        | dogfood_compute                 | r1           | ready                |          0.2168367 |  0.1701831817626953 |              0.5 |              1 | compute
 u92        | u82        | compute_instruqt                | r1           | ready                |          2.7085386 |  0.4721641540527344 |              0.5 |              1 | compute
 u74        | u65        | pggcptest_public_pg             | linked       | ready                |          1.2607948 |  0.6923675537109375 |             0.25 |              1 | source
 u88        | u78        | materialize_public_load_gen     | linked       | ready                | 3.1742166000000003 |  0.8412361145019531 |             0.25 |              1 | source
 u78        | u69        | materialize_public_json_sink    | linked       | ready                |            0.32444 |  0.3037452697753906 |             0.25 |              1 | compute
 u79        | u70        | materialize_public_public_items | linked       | ready                |          1.6369404 |  0.6608009338378906 |             0.25 |              1 | source
(12 rows)
```

Might some magic happen in the background?

Reply:
Hum. It seems like a sink with a linked cluster is getting marked as `compute`:

```
 u78        | u69        | materialize_public_json_sink    | linked       | ready                |            0.32444 |  0.3037452697753906 |             0.25 |              1 | compute
```

Also, I wonder what happens when you have multiple sources **or** sources and sinks in the same cluster. I think we could leave this out, to avoid surprises!

Reply:
That cluster gave me a big headache this week. It appears that due to a bug related [to this issue](https://github.com/MaterializeInc/materialize/issues/17014). ([Slack thread](https://materializeinc.slack.com/archives/CU7ELJ6E9/p1685122339537289))